### PR TITLE
[AppConfig] Default to false if null returned when deserilizing ENABLED in feature flag setting

### PR DESCRIPTION
**Pull request by mssfang**
# Description
When the ENABLED is null, use the default value false.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by mrm9084** [comment]
This should fix the pipeline for https://github.com/Azure/azure-sdk-for-java/pull/34022

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationSettingDeserializationHelper.java [hunk 1]
```diff
@@ -181,30 +181,31 @@ private static FeatureFlagConfigurationSetting getFeatureFlagPropertyValue(JsonP
...
         if (token == JsonToken.FIELD_NAME && DESCRIPTION.equals(parser.getCurrentName())) {
             token = parser.nextToken();
             if (token == JsonToken.VALUE_STRING) {
                 map.put(DESCRIPTION, parser.getText());
             }
+            token = parser.nextToken();
         }
         // display name
-        token = parser.nextToken();
         if (token == JsonToken.FIELD_NAME && DISPLAY_NAME.equals(parser.getCurrentName())) {
```
Comment (mrm9084) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationSettingDeserializationHelper.java:195]: This field is optional. The value of token here is `enabled`. This resulted line 202 skipping the `enabled` and getting the `VALUE_TRUE`/`VALUE_FALSE`, so it is never set.

**Comment by chenrujun** [comment]
Glad to see that [java - spring - tests](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2638237&view=results) passed.

The `java - appconfiguration - ci (Build Analyze)` failure is not caused by current PR. Link to related PR: https://github.com/Azure/azure-sdk-for-java/pull/33985#issuecomment-1473399515

**Review by chenrujun — APPROVED** [review]

**Pull request merged**